base/strings/string.jl

```diff
@@ -160,26 +167,24 @@ const utf8_trailing = [
 ## required core functionality ##
 
 function endof(s::String)
-    p = pointer(s)
     i = sizeof(s)
-    while i > 0 && is_valid_continuation(unsafe_load(p,i))
+    while i > 0 && is_valid_continuation(unsafe_load(s, i))
         i -= 1
     end
     i
 end
 
 function length(s::String)
-    p = pointer(s)
     cnum = 0
     for i = 1:sizeof(s)
-        cnum += !is_valid_continuation(unsafe_load(p,i))
+        cnum += !is_valid_continuation(unsafe_load(s, i))
     end
     cnum
 end
 
-@noinline function slow_utf8_next(p::Ptr{UInt8}, b::UInt8, i::Int, l::Int)
```

Comment:
The `noinline` can probably be removed?

Reply:
I think that was there for performance reasons to keep the slow path out of the icache?

Reply:
OK, sure. (Not that the current heuristic inlines the big function anyway....)